### ISSUE TITLE
SIP projection 

### ISSUE THREAD

**Issue by cnaw** [astromatic/sextractor#18]
Would it be possible to include the SIP convention as part of the valid WCS transformations ?
It adds non-linear terms to the tangent plane approximation and is described in this reference:

https://ui.adsabs.harvard.edu/abs/2005ASPC..347..491S/abstract

Thanks!

Christopher Willmer
Steward Observatory

**Comment by kirxkirx** [comment]
That would be very useful. For now I use an ugly workaround: measure XY star positions with SExtractor and then convert them to sky coordinates with `xy2sky` from WCSTools (that does support SIP).

**Comment by Eririf** [comment]
Source Extractor  version 2.25.0 still don't support SIP
I think wcslib has already support SIP :)

**Comment by ebertin** [comment]
Yes, SExtractor needs to switch to a much more recent of wcslib. This may be a significant undertaking though, as it is currently using patched code from a 20+ year old version :smile:. I'll do my best with the time I have!

**Comment by rmathar** [comment]
Switching in wcslib 8.4 is doable. It means deleting the `wcs` subdirectory,
adding `wcs.h` detection to `configure.ac` and substituting `wcsset`, `wcsfwd`
and `wcsrev` calls in `fitswcs.c` by mixes of `wcsini`, `wcssub`, `wcss2p` and
`wcsp2s` calls. Some kind of experimental version on this is in the `skymaker` fork
in https://github.com/rmathar/skymaker . However, this would not enable any support of SIP:
most of the FITS header parsing in the `read_wcs` function, for example, is not done
by examining the FITS cards by means of the wcslib , but by some manual processing.
